rfd/0017-sign-everything.md

```diff
@@ -0,0 +1,32 @@
+# Sign Everything
+
+Require [signed commits](https://docs.github.com/en/free-pro-team@latest/github/authenticating-to-github/about-commit-signature-verification) 
+for protected branches `master` and `branch/x.y`.
+
+Sign all files published to Github releases.
+
+## Why 
+
+Have a audit trail of all commits in the system and prevent commit author impersonation.
+
+## How
+
+Turn on commit signing for users and bots following steps 
+[here](https://docs.github.com/en/free-pro-team@latest/github/authenticating-to-github/about-commit-signature-verification)
+
+**Teleport specific key repository**
+
+Establish Teleport-specific repository of supported public keys used by core team developers and have an 
+additional verification step done by Drone pipeline.
+
+Developers should register [Yubikeys](https://www.engineerbetter.com/blog/yubikey-signed-commits/)
+as used keys.
+
+**Support for external contributors**
+
+Once pull request from external contributor has been approved, Drone can amend the commit, sign it using our CI/CD key
+and rebase to master.
+
+**Remove the github releases files**
```

Comment:
I'm not sure we can do this, I think it's automatic on their end. We can definitely publish source artifacts with checksums though.

Reply:
We can remove the tar.gz files after we create a release and update with our own. Then github will not alter and generate the tarballs.